src/collection.rs

```diff
@@ -41,11 +51,17 @@ impl QueryParamsRaw {
         let skip = self.skip.map(|v| v as usize);
         let top = self.top.map(|v| v as usize);
 
+        // XXX consider returning an error here ?
+        let filter = self
+            .filter
+            .and_then(|f| odata_params::filters::parse_str(f).ok());
```

Comment:
I think we definitely should return an error if parsing fails (see my comment above) as we don't want user to think that filtering worked fine while in fact it was completely ignored